server/src/main/java/org/apache/iotdb/db/monitor/StatMonitor.java

```diff
@@ -16,317 +16,291 @@
+    if (storageGroupNames.isEmpty()) {
+      return;
+    }
     try {
-      for (Map.Entry<String, String> entry : hashMap.entrySet()) {
-        if (entry.getValue() == null) {
-          logger.error("Registering metadata but data type of {} is null", entry.getKey());
-        }
-
-        if (!mManager.isPathExist(new PartialPath(entry.getKey()))) {
-          mManager.createTimeseries(new PartialPath(entry.getKey()), TSDataType.valueOf(entry.getValue()),
-              TSEncoding.valueOf("RLE"),
-              TSFileDescriptor.getInstance().getConfig().getCompressor(),
-              Collections.emptyMap());
+      for (PartialPath storageGroupName : storageGroupNames) {
+        if (!storageGroupName.equals(MonitorConstants.STAT_STORAGE_GROUP_NAME)) {
+          // for storage group which is not global, only TOTAL_POINTS is registered now
+          PartialPath fullPath = new PartialPath(MonitorConstants.STAT_STORAGE_GROUP_ARRAY)
+              .concatNode("\"" + storageGroupName + "\"")
+              .concatNode(StatMeasurementConstants.TOTAL_POINTS.getMeasurement());
+          registSeriesToMManager(fullPath);
+
+          List<PartialPath> seriesList = monitorSeriesMap
+              .computeIfAbsent(storageGroupName.toString(), k -> new ArrayList<>());
+          seriesList.add(fullPath);
+          cachedValueMap.putIfAbsent(fullPath, (long) 0);
         }
       }
     } catch (MetadataException e) {
       logger.error("Initialize the metadata error.", e);
     }
   }
 
-  public void recovery() {
-    // // restore the FildeNode Manager TOTAL_POINTS statistics info
+  private void registSeriesToMManager(PartialPath fullPath) throws MetadataException {
+    if (!mManager.isPathExist(fullPath)) {
+      mManager.createTimeseries(fullPath,
+          TSDataType.valueOf(MonitorConstants.INT64),
+          TSEncoding.valueOf("RLE"),
+          TSFileDescriptor.getInstance().getConfig().getCompressor(),
+          Collections.emptyMap());
+    }
   }
 
-  void activate() {
-    service = IoTDBThreadPoolFactory.newScheduledThreadPool(1,
-        ThreadName.STAT_MONITOR.getName());
-    service.scheduleAtFixedRate(
-        new StatBackLoop(), 1, backLoopPeriod, TimeUnit.SECONDS);
+  public void updateStatStorageGroupValue(String storageGroupName, int successPointsNum) {
+    List<PartialPath> monitorSeries = monitorSeriesMap.get(storageGroupName);
+    // update TOTAL_POINTS of each storage group
+    cachedValueMap.computeIfPresent(monitorSeries.get(0),
+        (key, oldValue) -> oldValue + successPointsNum);
   }
 
-  void clearIStatisticMap() {
-    statisticMap.clear();
+  public void updateStatGlobalValue(int successPointsNum) {
+    List<PartialPath> monitorSeries = monitorSeriesMap
+        .get(MonitorConstants.STAT_STORAGE_GROUP_NAME);
+    for (int i = 0; i < monitorSeries.size() - 1; i++) {
+      // 0 -> TOTAL_POINTS, 1 -> REQ_SUCCESS, 2 -> REQ_FAIL
```

Comment:
？？ if so, why use `for`?
and how about `REQ_FAIL`

Reply:
I designed it for ... extensibility :D. Maybe we will add more metrics later...  I will fix it~ 
`REQ_FAIL` is not calculated with the successful situation.